src/Farmer/Builders/Builders.LogAnalytics.fs

```diff
@@ -0,0 +1,80 @@
+﻿[<AutoOpen>]
+module Farmer.Builders.LogAnalytics
+open Farmer
+open Farmer.LogAnalytics
+open Farmer.Arm.LogAnalytics
+type  WorkSpaceconfig =
+    { Name: ResourceName
+      Sku: Sku
+      retentionInDays: int option
+      publicNetworkAccessForIngestion: string option
+      publicNetworkAccessForQuery: string option
+      Tags: Map<string,string>
+      }
+    interface IBuilder with
+        member this.DependencyName = this.Name
+        member this.BuildResources location =
+            [ { Name = this.Name
+                Location = location
+                Sku = this.Sku
+                retentionInDays = 
+                    match this.retentionInDays with
+                        |Some value ->
+                            match this.Sku with
+                                |Standard   when value<>30 -> failwithf "the retention Days for Standard must be 30."
+                                |Premium    when  value<>365 -> failwithf "the retention Days for Premium must be 365."
```

Comment:
Consider flipping the match to a tupled match to remove the need for `when` clauses e.g.

```fsharp
match this.Sku, this.RetentionInDays with
| Standard, Some 30 -> ...
| Standard, Some _ -> failwith "must be 30 days"
```

etc.